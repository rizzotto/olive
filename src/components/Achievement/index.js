import React from 'react';
import useStyles from './styles';
import PropTypes from 'prop-types';
import LinearProgressWithLabel from '../../components/LinearProgressWithLabel';
import { ReactComponent as Trophy } from '../../assets/trophy.svg';
import { ReactComponent as TrophyPlatin } from '../../assets/trophyPlatin.svg';
import { Grow } from '@material-ui/core';
// import { useState } from 'react';

function Achievement({ disabled, total, progress, main, tag, ...others }) {
  const classes = useStyles({ disabled });
  // Future animation
  // const [value, setValue] = useState();

  // useEffect(() => {
  //   setValue(progress);
  // }, []);

  return (
    <Grow in>
      <div className={classes.container} {...others}>
        {main ? (
          <TrophyPlatin style={{ filter: disabled && 'grayscale(1)' }} />
        ) : (
          <Trophy style={{ filter: disabled && 'grayscale(1)' }} />
        )}
        <div className={classes.content}>
          {tag ? (
            <div className={classes.title}>Receitas com a tag {tag}</div>
          ) : (
            <div className={classes.title}>Complete {total} Receitas</div>
          )}
          <LinearProgressWithLabel
            achievement
            disabled={disabled}
            value={progress}
            total={total}
          />
        </div>
      </div>
    </Grow>
  );
}

Achievement.propTypes = {
  disabled: PropTypes.bool,
  isButton: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  selected: PropTypes.bool,
};

Achievement.defaultProps = {
  disabled: false,
  isButton: false,
  onClick: () => {},
  selected: false,
};

export default Achievement;
