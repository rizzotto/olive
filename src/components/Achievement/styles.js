import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  container: {
    alignItems: 'center',
    backgroundColor: theme.palette.grey.main,
    borderRadius: 12,
    display: 'flex',
    height: 81,
    justifyContent: 'flex-start',
    padding: 10,
    width: '100%',
  },
  content: {
    width: '100%',
    paddingLeft: 10,
    textAlign: 'left',
  },
  title: {
    color: props => (props.disabled ? theme.palette.grey.darker : '#000'),
    fontWeight: 500,
    paddingBottom: 5,
  },
}));

export default useStyles;
