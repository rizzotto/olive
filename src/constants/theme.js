import { createMuiTheme } from '@material-ui/core/styles';

const primaryGradient = gradient => {
  return `rgba(53, 119, 53, ${gradient})`;
};

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryGradient(1),
      gradient80: primaryGradient(0.8),
      gradient60: primaryGradient(0.6),
      gradient40: primaryGradient(0.4),
      gradient20: primaryGradient(0.2),
      gradient10: primaryGradient(0.1),
    },
    base: {
      primary: '#FEF9EE',
      secondary: '#F6EDD9',
      tertiary: '#EFEEE5',
    },
    secondary: {
      main: '#52944F',
    },
    tertiary: {
      main: '#6FB26B',
    },
    quaternary: {
      main: '#8DD187',
    },
    grey: {
      main: '#F3F3F3',
      medium: '#DCDCDD',
      darker: '#8A8A8E',
    },
    contrastThreshold: 3,
    tonalOffset: 0.2,
  },
  typography: {
    fontFamily: 'Montserrat, Helvetica, Arial, sans-serif',
    color: '#212121',
  },
});

export default theme;
