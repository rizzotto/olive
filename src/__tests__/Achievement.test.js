import React from 'react';
import renderer from 'react-test-renderer';
import Achievement from '../components/Achievement';
import { ThemeProvider } from '@material-ui/core';
import theme from '../constants/theme';

test('renders a simple Achievement', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Achievement />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
