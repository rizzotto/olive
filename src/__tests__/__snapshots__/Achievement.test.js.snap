// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a simple Achievement 1`] = `
<div
  className="makeStyles-container-1"
  isButton={false}
  onClick={[Function]}
  selected={false}
>
  <svg>
    trophy.svg
  </svg>
  <div
    className="makeStyles-content-2"
  >
    <div
      className="makeStyles-title-3"
    >
      Complete 
       receitas
    </div>
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <div
        className="MuiBox-root MuiBox-root-6"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="MuiLinearProgress-root makeStyles-root-4 MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
            style={
              Object {
                "transform": "translateX(NaN%)",
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-7"
      >
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          NaN/undefined
        </p>
      </div>
    </div>
  </div>
</div>
`;
